src/main.rs

```diff
@@ -495,15 +495,12 @@ impl App {
         let config = self.build_stub_config(args)?;
 
         let stub_generator = match args.get_one::<PathBuf>("from-file") {
-            Some(fname) => {
-                if fname.to_str() == Some("-") {
-                    let mut input = String::new();
-                    std::io::Read::read_to_string(&mut std::io::stdin(), &mut input)?;
-                    input
-                } else {
-                    std::fs::read_to_string(fname)?
-                }
+            Some(fname) if fname.to_str() == Some("-") => {
```

Comment:
So `clash gen --from-file -` reads from stdin? I'm not aware of any convention that makes `-` mean `stdin`, are there any examples?

Reply:
> So `clash gen --from-file -` reads from stdin? I'm not aware of any convention that makes `-` mean `stdin`, are there any examples?

It's a POSIX utility syntax guideline [^1]. Most compilers and static tooling like linters and type checkers use it. For example the test script in #49 takes advantage of it with both rustc and gcc.

[^1]: [POSIX.1-2017](https://pubs.opengroup.org/onlinepubs/9699919799/), chapter 12.2 Utility Syntax Guidelines, guideline 13

Reply:
> I am not sure about the point of allowing this `from-file` to take input directly from `stdin`.

The point is that you can pipe the input into `clash` to check things without creating an unnecessary temporary file. It's especially useful when the input comes from another program. It's also only like 5 lines of extra code to implement so why not?.

Reply:
> It's a POSIX utility syntax guideline [1](#user-content-fn-1-4304eac3e4cdedcb23a3807b996c30bf). Most compilers and static tooling like linters and type checkers use it. For example the test script in #49 takes advantage of it with both rustc and gcc.
> ## Footnotes
> 
>     1. [POSIX.1-2017](https://pubs.opengroup.org/onlinepubs/9699919799/), chapter 12.2 Utility Syntax Guidelines, guideline 13 [↩](#user-content-fnref-1-4304eac3e4cdedcb23a3807b996c30bf)

Fair enough, I take back what I said :+1: